make/tsb.sh

```diff
@@ -36,43 +36,50 @@ function help() {
 #
 # Usage: deploy_mp
 function deploy_mp() {
+  print_info "Going to deploy tsb management plane"
   set -e
 
-  local cloud_provider=$(jq -r '.tsb_mp.cloud' "${TFVARS_JSON}")
-  print_info "Going to deploy tsb management plane on cloud '${cloud_provider}'"
-  source ${BASE_DIR}/k8s_auth.sh k8s_auth_${cloud_provider}
+  local cluster=$(get_mp_cluster_config "${TFVARS_JSON}")
+  local cloud=$(get_cluster_cloud "${cluster}")
+  local workspace=${NAME_PREFIX}
+  source "${BASE_DIR}/k8s_auth.sh" k8s_auth_${cloud}
+  echo processing cluster: "${cluster}"
 
   run "pushd tsb/mp > /dev/null"
-  run "terraform workspace select default"
+  run "terraform workspace new ${workspace} || true"
+  run "terraform workspace select ${workspace}"
   run "terraform init"
-  run "terraform apply ${TERRAFORM_APPLY_ARGS} -target=module.cert-manager -target=module.es -target=data.terraform_remote_state.infra -var-file=../../${TFVARS_JSON}"
-  run "terraform apply ${TERRAFORM_APPLY_ARGS} -target=module.tsb_mp.kubectl_manifest.manifests_certs -target=data.terraform_remote_state.infra -var-file=../../${TFVARS_JSON}"
-  run "terraform apply ${TERRAFORM_APPLY_ARGS} -var-file=../../${TFVARS_JSON}"
+  run "terraform apply ${TERRAFORM_APPLY_ARGS} -target=module.cert-manager -target=module.es -target=data.terraform_remote_state.infra -var-file=../../${TFVARS_JSON} -var=cluster='${cluster}'"
+  run "terraform apply ${TERRAFORM_APPLY_ARGS} -target=module.tsb_mp.kubectl_manifest.manifests_certs -target=data.terraform_remote_state.infra -var-file=../../${TFVARS_JSON} -var=cluster='${cluster}'"
+  run "terraform apply ${TERRAFORM_APPLY_ARGS} -var-file=../../${TFVARS_JSON} -var=cluster='${cluster}'"
   run "terraform output ${TERRAFORM_OUTPUT_ARGS} | jq . > ../../outputs/terraform_outputs/terraform-tsb-mp.json"
   run "terraform workspace select default"
   run "popd > /dev/null"
 
-  print_info "Finished deploying tsb management plane on cloud '${cloud_provider}'"
+  print_info "Finished deploying tsb management plane on cloud '${cloud}'"
 }
 
 # This function deploys tsb management plane fqdn on configured cloud provider.
 #
 # Usage: deploy_mp_fqdn
 function deploy_mp_fqdn() {
+  print_info "Going to deploy tsb management plane fqdn"
   set -e
 
-  local fqdn=$(jq -r '.tsb_fqdn' "${TFVARS_JSON}")
+  local cluster=$(get_mp_cluster_config "${TFVARS_JSON}")
+  local fqdn=${TETRATE_FQDN}
   local address=$(jq -r "if .ingress_ip.value != \"\" then .ingress_ip.value else .ingress_hostname.value end" outputs/terraform_outputs/terraform-tsb-mp.json)
-  local dns_provider=$(jq -r '.dns_provider' "${TFVARS_JSON}")
-  if [ "${dns_provider}" == "null" ]; then
-    dns_provider=$(jq -r '.tsb_fqdn' "${TFVARS_JSON}" | cut -d"." -f2 | sed 's/sandbox/gcp/g')
-  fi
-  print_info "Going to deploy tsb management plane fqdn on cloud '${dns_provider}'"
+  local dns_provider=$(get_cluster_cloud "${cluster}")
```

Comment:
have some scenarios with dns_provider will be different then fqdn provider, so will work to add logic similar to:
```sh
  local dns_provider=$(jq -r '.dns_provider' "${TFVARS_JSON}")
  if [ "${dns_provider}" == "null" ]; then
    dns_provider=$(jq -r '.tsb_fqdn' "${TFVARS_JSON}" | cut -d"." -f2 | sed 's/sandbox/gcp/g')
  fi
```
to variables.sh